tool/tsh/app.go

```diff
@@ -228,30 +228,42 @@ func onAppConfig(cf *CLIConf) error {
 	if err != nil {
 		return trace.Wrap(err)
 	}
-	conf, err := formatAppConfig(tc, profile, app.Name, app.PublicAddr, cf.Format, "")
+	conf, err := formatAppConfig(tc, profile, app.Name, app.PublicAddr, cf.Format, "", cf.InsecureSkipVerify)
 	if err != nil {
 		return trace.Wrap(err)
 	}
 	fmt.Print(conf)
 	return nil
 }
 
-func formatAppConfig(tc *client.TeleportClient, profile *client.ProfileStatus, appName, appPublicAddr, format, cluster string) (string, error) {
+func formatAppConfig(tc *client.TeleportClient, profile *client.ProfileStatus, appName, appPublicAddr, format, cluster string, insecure bool) (string, error) {
 	var uri string
 	if port := tc.WebProxyPort(); port == teleport.StandardHTTPSPort {
 		uri = fmt.Sprintf("https://%v", appPublicAddr)
 	} else {
 		uri = fmt.Sprintf("https://%v:%v", appPublicAddr, port)
 	}
-	curlCmd := fmt.Sprintf(`curl \
+
+	var curlCmd string
+	if insecure {
+		curlCmd = fmt.Sprintf(`curl \
   --cacert %v \
```

Comment:
Hmm, not sure - why do we still want to show this flag in insecure mode?

Reply:
@r0mamt,
My idea was to keep the curl command working if Teleport runs with an untrusted certificate. 
 If users are testing it locally it would be difficult for them to understand what is wrong since we gave them the command to run.

Reply:
Right, my question was how keeping --cacert flag is going to help since curl is still not going to be able to validate self-signed cert presented by Teleport. We can add `-k` flag to curl command when insecure flag is given, but we should also then output a warning saying that curl command will skip cert validation.

Reply:
I think the `-k` and the warning makes a lot of sense. @tigrato, I think I'll go ahead and implement that.

Reply:
Let me know what you think about https://github.com/gravitational/teleport/pull/16769#issuecomment-1262313995.